Source/WebCore/svg/SVGGraphicsElement.cpp

```diff
@@ -132,11 +132,15 @@ AffineTransform* SVGGraphicsElement::ensureSupplementalTransform()
 
 void SVGGraphicsElement::attributeChanged(const QualifiedName& name, const AtomString& oldValue, const AtomString& newValue, AttributeModificationReason attributeModificationReason)
 {
-    SVGTests::parseAttribute(name, newValue);
-    SVGElement::attributeChanged(name, oldValue, newValue, attributeModificationReason);
-
+    // Updating the SVG transform baseVal needs to happen before calling attributeChanged().
+    // attributeChanged() invokes svgAttributeChanged(), which calls RenderLayerModelObjects
+    // repaintOrRelayoutAfterSVGTransformChange() method, if LBSE is activated. To determine
+    // if the SVG transform changed, m_transform has to be updated before.
```

Comment:
I think the order of changing the SVG property member and calling base class attributeChanged() does matter for legacy layout as well. 

`SVGElement::attributeChanged()` need to be called last after changing the superclasses' members because it calls the virtual method `svgAttributeChanged()` which assumes the class members have been updated due to an attribute change.

So I think this comment is not 100% correct. The same problem did happen in many other places and we can't afford adding the same comment in these places. I would suggest adding a layout test which which changes the `transform` attribute dynamically.

Reply:
I am going to take care of it.